### PR TITLE
Test normalizeEmail and permissionResolve

### PR DESCRIPTION
**Pull request by travis-wb**

### CODE REVIEW

File: lib/node_core/src/normalize_email.test.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    const result = normalizeEmail('JOHNNY@CIDI.TEST');
+    assert.strictEqual(result, 'johnny@cidi.test');
+  });
+
+  it('strips punctuation from the user part only', () => {
+    let result = normalizeEmail('johnny.cash@cidi.test.testing');
+    assert.strictEqual(result, 'johnnycash@cidi.test.testing');
+
+    result = normalizeEmail('johnny.cash');
+    assert.strictEqual(result, 'johnny.cash');
```
Comment (travis-wb) [inline lib/node_core/src/normalize_email.test.ts:15]: @cliff-wb is this the right behaviour? If not let me know what I should assert and I can adjust the subject accordingly

File: lib/node_core/src/permission_resolve.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      {relation_type: 'remove', permission_name: 'permissionB'},
+      {relation_type: 'add_grant', permission_name: 'permissionC'},
+
+      {relation_type: 'remove', permission_name: 'permissionA'},
+    ]);
+
+    assert.deepStrictEqual(result, {
+      permissionC: true,
+      '+permissionC': true,
+    });
```
Comment (travis-wb) [inline lib/node_core/src/permission_resolve.test.ts:56]: If you include the same `permission_name` in the input array twice, the last one encountered wins. I'm not sure if this is desirable, or possible. I can look into it.

@cliff-wb I also wanted to ask you if the `debug` and `critical` calls to the debugger are important. Should be `Ctor`-izing the subject so I can pass in a debugger mock and assert that they're called?

File: lib/node_core/src/normalize_email.test.ts (hunk 1 above)
Comment (cliff-wb) [reply lib/node_core/src/normalize_email.test.ts:15]: It's fine. GIGO

Comment (cliff-wb) [reply lib/node_core/src/normalize_email.test.ts:15]: probably should test 'Foo+bar@baz.COM' and ensure it keeps the '+'

File: lib/node_core/src/permission_resolve.test.ts (hunk 2 above)
Comment (cliff-wb) [reply lib/node_core/src/permission_resolve.test.ts:56]: The query that feeds this should not get dups; that should be added as an expectation in a comment  in the function.
I wouldn't bother testing debug statements.

File: lib/node_core/src/normalize_email.test.ts (hunk 1 above)
Comment (travis-wb) [reply lib/node_core/src/normalize_email.test.ts:18]: There are now tetst to over the `+` and a capitalized TLD.

**Review by cliff-wb — APPROVED** [review]

**Pull request merged**